src/components/AddItem.jsx

```diff
@@ -24,37 +25,73 @@ function AddItem() {
   const [itemName, setItemName] = useState('');
   const [frequency, setFrequency] = useState(7);
   const [notification, setNotification] = useState('');
+  const [list, setList] = useState([]);
+  const localToken = localStorage.getItem('list-token');
+  const [duplicateMessage, setDuplicateMessage] = useState(null);
 
-  //retrive the token from localStorage
-  const token = localStorage.getItem('list-token');
+  //useEffect to setList of items in that user's list which will be used for duplicate comparison
+  useEffect(() => {
+    const unsubscribe = onSnapshot(collection(db, localToken), (snapshot) => {
+      const snapshotDocs = [];
+      snapshot.forEach((doc) => snapshotDocs.push(doc.data()));
+      setList(snapshotDocs);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, [localToken]);
+
+  function duplicateCheck(itemName, list) {
```

Comment:
the idea for refactoring: we can have a separate function to get rid of punctuation outside of component and use it when it's needed just to make code more clear. for example
```
const removePunctuation = (string) => {
  const punctuationlessString = string.toLowerCase()
     .replace(/[.,\/#!$%\^&\*;:@{}=\-_`~()]/g, '')
     .replace(/\s{2,}/g, '')

return punctuationlessString
}
```

Reply:
Agreed - this could be a function exported from a utils.js file (or whatever we want to call it)

Reply:
I also think the logic for cleaning up user input should happen earlier than here. When I type in 'coffee' as an itemName, that is saved to our state which is fine. But then if I try 'coffee!!!!!', that is what we get in state as well. The validation check works since it isn't getting saved to the db, but I think it might be worth considering sanitizing the input and setting that as itemName in state rather than something with a bunch of extra characters.

Reply:
As an example, I was able to save this to the db without a problem.

<img width="292" alt="Screen Shot 2022-01-28 at 2 20 45 PM" src="https://user-images.githubusercontent.com/78327387/151629508-63a8ccbb-fa68-46ca-a04f-2ba92a4ebdfb.png">